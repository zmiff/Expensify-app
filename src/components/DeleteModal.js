import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startRemoveExpense } from '../actions/expenses';
import { closeModal }  from '../actions/modal';

export class DeleteModal extends React.Component {
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.closeModal();
    this.props.history.push('/');
  };
  onClose = () => {
    this.props.closeModal();
  }
  render() {
    return (
      <Modal
        ariaHideApp={false}
        shouldCloseOnOverlayClick={true}
        isOpen={ this.props.isOpen }
        contentLabel="Delete Expense"
        closeTimeoutMS={200}
        className="modal"
      >
        <h3 className="modal__title">Confirm Delete Expense</h3>
        <button className="button" onClick={ this.onClose } >Close</button>
        <button className="button--danger" onClick={ this.onRemove } >Delete Expense</button>
      </Modal>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    isOpen: state.modal.isOpen
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
  closeModal: () => dispatch(closeModal())
});



export default connect(mapStateToProps, mapDispatchToProps)(DeleteModal);
