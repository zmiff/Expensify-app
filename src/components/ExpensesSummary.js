import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import configureStore from '../store/configureStore';
import selectExpenses from '../selectors/expenses';
import SelectExpenesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpenseTotal = numeral(expensesTotal / 100).format('0,0.00')+'kr '

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Vieving <span>{expensesCount}</span> {expenseWord} totaling: <span>{formattedExpenseTotal}</span></h1>
        <div className="page-header__actions">
          <Link to="/create" className="button">Add Expense</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
    let expenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesCount: expenses.length,
    expensesTotal: SelectExpenesTotal(expenses)
  };
}

export default connect(mapStateToProps)(ExpensesSummary);
