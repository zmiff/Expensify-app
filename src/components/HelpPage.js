import React from 'react';

const HelpPage = () => (
  <div>
    This is from HelpPage component
  </div>
);

export default HelpPage;
