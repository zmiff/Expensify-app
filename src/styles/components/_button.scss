.button {
  background-color: $light-blue;
  border: none;
  border-radius: 2px;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight:300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--secondary {
  @extend .button;
  background-color: $light-grey;
}

.button--link {
  background: none;
}
