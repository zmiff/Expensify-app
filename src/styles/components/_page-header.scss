.page-header {
  background: $off-white;
  margin-bottom: $l-size;
  padding: $l-size 0;
}

.page-header__Actions {
  margin-top: $m-size;
}

.page-header__title {
  font-weight:300;
  margin: 0;
  span {
    font-weight: 700;
  }
}

.page-header__under--title {
  padding: 0;
  margin: 0 0 $m-size 0;
  font-size: $font-size-med;
  font-weight: 100;

}
