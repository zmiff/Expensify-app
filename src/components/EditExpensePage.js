import React from 'react';
import {connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import DeleteModal from './DeleteModal';
import { startEditExpense } from '../actions/expenses';
import { openModal }  from '../actions/modal';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={ this.onSubmit }
          />
          <button className="button--secondary" onClick={ () => {this.props.openModal()} }>Remove Expense</button>
        </div>
        <div>
          <DeleteModal
            expense={ this.props.expense }
            history={ this.props.history }
           />
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
    isOpen: state.modal.isOpen
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  openModal: () => dispatch(openModal())
});


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
