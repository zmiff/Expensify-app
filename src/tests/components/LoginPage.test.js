import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage correctly', () => {
  const wrapper = shallow(<LoginPage startLogin={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin in button click', () => {
  const onClickSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={ onClickSpy } />);
  wrapper.find('button').simulate('click', () => {
    expect(onClickSpy).toHaveBeenCalled();
  })
});
