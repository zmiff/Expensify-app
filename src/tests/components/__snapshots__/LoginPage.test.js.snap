// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
